### PR TITLE
HWS: Show dump statistics

### PR DESCRIPTION
**Pull request by hamdanigbaria**
Add "-s" flag, statistics, which will show dump statistics. For now we show STE's distribution in the table, this gives an overview about the table collisions.

Signed-off-by: Hamdan Igbaria <hamdani@nvidia.com>

### CODE REVIEW

File: hws/README.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ Running syntax
 | -pid DPDK_PID | Trigger DPDK app <PID> |
 | -port DPDK_PORT | Trigger DPDK app <PORT> (must provide PID with -pid) |
 | -extra_hw_res [pat, arg] | Request extra HW resources to be dumped. For example: -extra_hw_res pat,arg |
+| -s | Show dump statistics. |
```
Comment (kliteyn) [inline hws/README.md:50]: "Show dump statistics" - what does it mean? Maybe "Dump steering statistics"?
Also, some additional info on what is dumped is welcome.
E.g.: "Dump steering statistics, such as bla and bla"

File: hws/src/dr_dump_hw.py [hunk 2]
```diff
@@ -59,6 +60,11 @@ def parse_fw_ste_rd_bin_output(fw_ste_index, load_to_db, file):
         if data[2:8] == RESOURCE_DUMP_SEGMENT_TYPE_STE_BIN:
             ste = '0x' + data[32:]
             hit_add = ste[32 : 41]
+            if first_ste:
+                ste_addr = '0x' + data[16:24]
+                if ste_addr < min_addr:
+                    min_addr = ste_addr
+                first_ste = False
```
Comment (kliteyn) [inline hws/src/dr_dump_hw.py:67]: in that case, you don't need lines 77-78 in the new code

File: hws/src/dr_matcher.py [hunk 3]
```diff
@@ -225,7 +272,6 @@ def dump_str(self, verbosity):
 
         return dump_obj_str(_keys, self.data)
 
-
```
Comment (kliteyn) [inline hws/src/dr_matcher.py:228]: remove

**Review by kliteyn — CHANGES_REQUESTED** [review]
please see comments

File: hws/README.md (hunk 1 above)
Comment (hamdanigbaria) [reply hws/README.md:50]: i think show dump statistics more suitable as we show stats about the dump itself. but added more info

**Comment by hamdanigbaria** [comment]
moved to new one, since it github blocked me from pushing again here. https://github.com/Mellanox/mlx_steering_dump/pull/76

**Pull request closed without merging**